### PR TITLE
chore: improve the stability of mysql

### PR DESCRIPTION
**Pull request by imbajin**
due to some network or time reason, the mysql ci  may fail frequently, now try to reconnect mysql server auto & find the 
root cause 

BTW, we should open the reconnect option in test mode by default

### CODE REVIEW

File: hugegraph-mysql/src/main/java/com/baidu/hugegraph/backend/store/mysql/MysqlOptions.java [hunk 1]
```diff
@@ -98,10 +98,20 @@ public static synchronized MysqlOptions instance() {
                     "false"
             );
 
+    public static final ConfigOption<Boolean> AUTO_RECONNECT =
+            new ConfigOption<>(
+                    "jdbc.auto_reconnect",
```
Comment (javeme) [inline hugegraph-mysql/src/main/java/com/baidu/hugegraph/backend/store/mysql/MysqlOptions.java:103]: prefer to rename to force_auto_reconnect, and set autoReconnect=true in MysqlSessions.buildUri() if force_auto_reconnect=true

File: hugegraph-mysql/src/main/java/com/baidu/hugegraph/backend/store/mysql/MysqlOptions.java [hunk 2]
```diff
@@ -98,10 +98,20 @@ public static synchronized MysqlOptions instance() {
...
+                    "Whether to auto reconnect mysql server if it meet " +
```
Comment (javeme) [inline hugegraph-mysql/src/main/java/com/baidu/hugegraph/backend/store/mysql/MysqlOptions.java:104]: Whether to forced auto reconnect to the database even if the connection fails at the first time.

**Comment by javeme** [comment]
seems need to reconnect when the database is running in an unstable container environment: https://stackoverflow.com/questions/13950496/what-is-java-io-eofexception-message-can-not-read-response-from-server-expect

**Comment by javeme** [comment]
```java
Error:  Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 183.058 s <<< FAILURE! - in com.baidu.hugegraph.core.CoreTestSuite
Error:  com.baidu.hugegraph.core.CoreTestSuite  Time elapsed: 183.042 s  <<< ERROR!
com.baidu.hugegraph.exception.ConnectionException: Failed to connect to MySQL
	at com.baidu.hugegraph.core.CoreTestSuite.init(CoreTestSuite.java:68)
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLNonTransientConnectionException: Could not create connection to database server. Attempted reconnect 10 times. Giving up.
	at com.baidu.hugegraph.core.CoreTestSuite.init(CoreTestSuite.java:68)
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Unknown database 'hugegraph'
	at com.baidu.hugegraph.core.CoreTestSuite.init(CoreTestSuite.java:68)

Error:  com.baidu.hugegraph.core.CoreTestSuite  Time elapsed: 183.058 s  <<< ERROR!
com.baidu.hugegraph.exception.ConnectionException: Failed to connect to MySQL
	at com.baidu.hugegraph.core.CoreTestSuite.clear(CoreTestSuite.java:80)
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLNonTransientConnectionException: Could not create connection to database server. Attempted reconnect 10 times. Giving up.
	at com.baidu.hugegraph.core.CoreTestSuite.clear(CoreTestSuite.java:80)
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Unknown database 'hugegraph'
	at com.baidu.hugegraph.core.CoreTestSuite.clear(CoreTestSuite.java:80)

Error: -29 08:05:02 [main] [ERROR] c.b.h.c.BaseCoreTest - Error when close()
com.baidu.hugegraph.exception.ConnectionException: Failed to connect to MySQL
	at com.baidu.hugegraph.backend.store.mysql.MysqlStore.open(MysqlStore.java:128) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.AbstractTransaction.<init>(AbstractTransaction.java:82) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.IndexableTransaction.<init>(IndexableTransaction.java:30) ~[classes/:?]
	at com.baidu.hugegraph.backend.tx.SchemaTransaction.<init>(SchemaTransaction.java:75) ~[classes/:?]
	at com.baidu.hugegraph.backend.cache.CachedSchemaTransaction.<init>(CachedSchemaTransaction.java:54) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph.openSchemaTransaction(StandardHugeGraph.java:406) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph.access$2600(StandardHugeGraph.java:111) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph$TinkerPopTransaction.getOrNewTransaction(StandardHugeGraph.java:1342) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph$TinkerPopTransaction.doOpen(StandardHugeGraph.java:1250) ~[classes/:?]
	at org.apache.tinkerpop.gremlin.structure.util.AbstractTransaction.open(AbstractTransaction.java:95) ~[gremlin-core-3.4.3.jar:3.4.3]
	at org.apache.tinkerpop.gremlin.structure.Transaction$READ_WRITE_BEHAVIOR$1.accept(Transaction.java:207) ~[gremlin-core-3.4.3.jar:3.4.3]
	at org.apache.tinkerpop.gremlin.structure.Transaction$READ_WRITE_BEHAVIOR$1.accept(Transaction.java:204) ~[gremlin-core-3.4.3.jar:3.4.3]
	at org.apache.tinkerpop.gremlin.structure.util.AbstractThreadLocalTransaction.doReadWrite(AbstractThreadLocalTransaction.java:92) ~[gremlin-core-3.4.3.jar:3.4.3]
	at org.apache.tinkerpop.gremlin.structure.util.AbstractTransaction.readWrite(AbstractTransaction.java:131) ~[gremlin-core-3.4.3.jar:3.4.3]
	at com.baidu.hugegraph.StandardHugeGraph.graphTransaction(StandardHugeGraph.java:483) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph.access$800(StandardHugeGraph.java:111) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph$StandardHugeGraphParams.initialized(StandardHugeGraph.java:1112) ~[classes/:?]
	at com.baidu.hugegraph.task.ServerInfoManager.removeSelfServerInfo(ServerInfoManager.java:337) ~[classes/:?]
	at com.baidu.hugegraph.task.ServerInfoManager.close(ServerInfoManager.java:121) ~[classes/:?]
	at com.baidu.hugegraph.task.StandardTaskScheduler.close(StandardTaskScheduler.java:495) ~[classes/:?]
	at com.baidu.hugegraph.task.TaskManager.closeScheduler(TaskManager.java:111) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph.close(StandardHugeGraph.java:888) ~[classes/:?]
	at com.baidu.hugegraph.core.CoreTestSuite.clear(CoreTestSuite.java:83) ~[classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_312]
```

**Comment by javeme** [comment]
```java
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running com.baidu.hugegraph.core.CoreTestSuite
2021-12-26 15:51:52 [main] [WARN] c.b.h.c.HugeConfig - The config option 'jdbc.sslmode' is redundant, please ensure it has been registered
2021-12-26 15:51:52 [main] [WARN] c.b.h.c.HugeConfig - The config option 'expired.delete_batch' is redundant, please ensure it has been registered
2021-12-26 15:51:52 [main] [WARN] c.b.h.c.HugeConfig - The config option 'test.tinkerpop.filter' is redundant, please ensure it has been registered
2021-12-26 15:51:53 [main] [INFO] c.b.h.b.s.m.MysqlStore - Connect to the jdbc url: 'jdbc:mysql://127.0.0.1:3306/hugegraph?useSSL=false&characterEncoding=utf-8&rewriteBatchedStatements=true&useServerPrepStmts=false&autoReconnect=true&maxReconnects=10&initialTimeout=5'
java.lang.Exception: just-for-debug
	at com.baidu.hugegraph.backend.store.mysql.MysqlSessions.connect(MysqlSessions.java:278) ~[classes/:?]
	at com.baidu.hugegraph.backend.store.mysql.MysqlSessions.open(MysqlSessions.java:217) ~[classes/:?]
	at com.baidu.hugegraph.backend.store.mysql.MysqlSessions.open(MysqlSessions.java:83) ~[classes/:?]
	at com.baidu.hugegraph.backend.store.mysql.MysqlStore.open(MysqlStore.java:124) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph.clearBackend(StandardHugeGraph.java:341) ~[classes/:?]
	at com.baidu.hugegraph.core.CoreTestSuite.init(CoreTestSuite.java:68) ~[classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_312]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_312]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_312]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_312]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50) ~[junit-4.12.jar:4.12]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) ~[junit-4.12.jar:4.12]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47) ~[junit-4.12.jar:4.12]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24) ~[junit-4.12.jar:4.12]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) ~[junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363) ~[junit-4.12.jar:4.12]
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365) ~[surefire-junit4-2.20.jar:2.20]
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272) ~[surefire-junit4-2.20.jar:2.20]
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:236) ~[surefire-junit4-2.20.jar:2.20]
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159) ~[surefire-junit4-2.20.jar:2.20]
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386) ~[surefire-booter-2.20.jar:2.20]
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323) ~[surefire-booter-2.20.jar:2.20]
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143) ~[surefire-booter-2.20.jar:2.20]
2021-12-26 15:52:38 [main] [INFO] c.b.h.b.s.m.MysqlStore - Connect to the jdbc url: 'jdbc:mysql://127.0.0.1:3306/hugegraph?useSSL=false&characterEncoding=utf-8&rewriteBatchedStatements=true&useServerPrepStmts=false&autoReconnect=true&maxReconnects=10&initialTimeout=5'
java.lang.Exception: just-for-debug
	at com.baidu.hugegraph.backend.store.mysql.MysqlSessions.connect(MysqlSessions.java:278) ~[classes/:?]
	at com.baidu.hugegraph.backend.store.mysql.MysqlSessions.open(MysqlSessions.java:217) ~[classes/:?]
	at com.baidu.hugegraph.backend.store.mysql.MysqlSessions.open(MysqlSessions.java:83) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph$TinkerPopTransaction.getOrNewTransaction(StandardHugeGraph.java:1346) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph$TinkerPopTransaction.doOpen(StandardHugeGraph.java:1255) ~[classes/:?]
	at org.apache.tinkerpop.gremlin.structure.util.AbstractTransaction.open(AbstractTransaction.java:95) ~[gremlin-core-3.4.3.jar:3.4.3]
	at org.apache.tinkerpop.gremlin.structure.Transaction$READ_WRITE_BEHAVIOR$1.accept(Transaction.java:207) ~[gremlin-core-3.4.3.jar:3.4.3]
	at org.apache.tinkerpop.gremlin.structure.Transaction$READ_WRITE_BEHAVIOR$1.accept(Transaction.java:204) ~[gremlin-core-3.4.3.jar:3.4.3]
	at org.apache.tinkerpop.gremlin.structure.util.AbstractThreadLocalTransaction.doReadWrite(AbstractThreadLocalTransaction.java:92) ~[gremlin-core-3.4.3.jar:3.4.3]
	at org.apache.tinkerpop.gremlin.structure.util.AbstractTransaction.readWrite(AbstractTransaction.java:131) ~[gremlin-core-3.4.3.jar:3.4.3]
	at com.baidu.hugegraph.StandardHugeGraph.graphTransaction(StandardHugeGraph.java:488) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph.access$800(StandardHugeGraph.java:111) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph$StandardHugeGraphParams.initialized(StandardHugeGraph.java:1117) ~[classes/:?]
	at com.baidu.hugegraph.task.ServerInfoManager.removeSelfServerInfo(ServerInfoManager.java:337) ~[classes/:?]
	at com.baidu.hugegraph.task.ServerInfoManager.close(ServerInfoManager.java:121) ~[classes/:?]
	at com.baidu.hugegraph.task.StandardTaskScheduler.close(StandardTaskScheduler.java:495) ~[classes/:?]
	at com.baidu.hugegraph.task.TaskManager.closeScheduler(TaskManager.java:111) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph.close(StandardHugeGraph.java:893) ~[classes/:?]
	at com.baidu.hugegraph.core.CoreTestSuite.clear(CoreTestSuite.java:83) ~[classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_312]
```

**Comment by javeme** [comment]
general error(no debug ):
```java
Error:  Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 3.056 s <<< FAILURE! - in com.baidu.hugegraph.core.CoreTestSuite
Error:  com.baidu.hugegraph.core.CoreTestSuite  Time elapsed: 3.049 s  <<< ERROR!
com.baidu.hugegraph.exception.ConnectionException: Failed to connect to MySQL
	at com.baidu.hugegraph.core.CoreTestSuite.init(CoreTestSuite.java:68)
Caused by: com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: 
Communications link failure

The last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server.
	at com.baidu.hugegraph.core.CoreTestSuite.init(CoreTestSuite.java:68)
Caused by: java.io.EOFException: Can not read response from server. Expected to read 4 bytes, read 0 bytes before connection was unexpectedly lost.
	at com.baidu.hugegraph.core.CoreTestSuite.init(CoreTestSuite.java:68)

Error:  com.baidu.hugegraph.core.CoreTestSuite  Time elapsed: 3.056 s  <<< ERROR!
com.baidu.hugegraph.exception.ConnectionException: Failed to connect to MySQL
	at com.baidu.hugegraph.core.CoreTestSuite.clear(CoreTestSuite.java:80)
Caused by: com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: 
Communications link failure

The last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server.
	at com.baidu.hugegraph.core.CoreTestSuite.clear(CoreTestSuite.java:80)
Caused by: java.io.EOFException: Can not read response from server. Expected to read 4 bytes, read 0 bytes before connection was unexpectedly lost.
	at com.baidu.hugegraph.core.CoreTestSuite.clear(CoreTestSuite.java:80)
```

File: hugegraph-mysql/src/main/java/com/baidu/hugegraph/backend/store/mysql/MysqlStore.java [hunk 3]
```diff
@@ -123,8 +123,8 @@ public synchronized void open(HugeConfig config) {
         try {
             this.sessions.open();
         } catch (Exception e) {
-            if (!e.getMessage().startsWith("Unknown database") &&
-                !e.getMessage().endsWith("does not exist")) {
+            LOG.error("Error to connect to mysql: {}", e.getMessage(), e);
+            if (!e.getMessage().startsWith("Unknown database")) {
```
Comment (javeme) [inline hugegraph-mysql/src/main/java/com/baidu/hugegraph/backend/store/mysql/MysqlStore.java:127]: "!e.getMessage().endsWith("does not exist")" is used by pgsql

**Comment by imbajin** [comment]
reopen it and replace action to `docker pull` only in mysql ci (not in all ci) or add a `if` condition 

another problem is mysql/pg start will failed sometimes, need check it

**Pull request closed without merging**